### PR TITLE
poc: skipToken type narrowing in useQuery

### PR DESCRIPTION
**Pull request by 0xApotheosis**
@gomesalexandre a little PoC on https://github.com/shapeshift/web/issues/6973.

As far as I can tell the type narrowing still does not work when the "enabled" filter is a `useMemo`.
It did work without it, but obviously we cannot do this.

### CODE REVIEW

File: src/pages/RFOX/components/Stake/StakeInput.tsx [hunk 1]
```diff
@@ -255,18 +255,22 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
...
-      feeAsset: feeAsset!, // see isGetStakeFeesEnabled
-      feeAssetMarketData: feeAssetMarketData!, // see isGetStakeFeesEnabled
-    }),
+    queryKey: ['stakeFees'],
+    queryFn: isGetStakeFeesEnabled
+      ? () => {
+          return reactQueries.common.evmFees({
+            to: RFOX_PROXY_CONTRACT_ADDRESS,
+            from: stakingAssetAccountId ? fromAccountId(stakingAssetAccountId).account : '',
+            accountNumber: stakingAssetAccountNumber,
```
Comment (0xApotheosis) [inline src/pages/RFOX/components/Stake/StakeInput.tsx:264]: Not type narrowed, still `number | undefined`.

File: src/pages/RFOX/components/Stake/StakeInput.tsx [hunk 2]
```diff
@@ -255,18 +255,22 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
...
+            data: callData,
```
Comment (0xApotheosis) [inline src/pages/RFOX/components/Stake/StakeInput.tsx:265]: Not type-narrowed, still ``0x${string}` | undefined`.

**Comment by 0xApotheosis** [comment]
Ah, I think there are some things to fix with the `@tanstack/react-query` bump.

**Comment by gomesalexandre** [comment]
See https://github.com/shapeshift/web/pull/7020 for follow-up on this

**Comment by 0xApotheosis** [comment]
We've implemented a solution for this, closing 🙏

**Pull request closed without merging**